src/goto-diff/unified_diff.cpp

```diff
@@ -275,7 +275,8 @@ unified_difft::differencest unified_difft::lcss(
       --i;
       --old_rit;
       --j;
-      --new_rit;
+      if(new_goto_program.instructions.begin()!=new_rit)
+        --new_rit;
```

Comment:
I am starting to wonder about the other cases of `new_rit` and `old_rit` being decremented, and do think that all of those will need the same adjustment.

Reply:
I would agree. I have a version with all other cases in this loop guarded. However, I have only PR the cases that actually error in test cases and as I have new test case I discovered new error. Would you like me to amend the commit with the complete version?

Reply:
1. Would it be possible to add those test cases somewhere?
2. Yes, I'd say add all those checks, this isn't such a huge performance risk.

Reply:
1. Possibly but it will take more time as they are on different repository.
2. Commit amended to have all cases in the loop checked.